### PR TITLE
ARROW-3630: [Plasma][GLib] Add GLib bindings of Plasma

### PR DESCRIPTION
**Pull request by shiro615**
It's GLib bindings of Plasma.

### CODE REVIEW

File: c_glib/plasma-glib/plasma-client.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+#pragma once
+
+#include <arrow-glib/gobject-type.h>
+
+G_BEGIN_DECLS
+
+#define GPLASMA_TYPE_PLASMA_CLIENT (gplasma_plasma_client_get_type())
+G_DECLARE_DERIVABLE_TYPE(GPlasmaPlasmaClient,
```
Comment (kou) [inline c_glib/plasma-glib/plasma-client.h]: Can you rename to `GPlasmaClient`? `PlasmaPlasma` is redundant.

File: c_glib/plasma-glib/plasma-client.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+ *
+ * Returns: A newly created #GPlasmaPlasmaClient.
+ *
+ * Since: 0.12.0
+ */
+GPlasmaPlasmaClient *
+gplasma_plasma_client_new(void)
+{
+  auto plasma_plasma_client = std::make_shared<plasma::PlasmaClient>();
+  return gplasma_plasma_client_new_raw(&plasma_plasma_client);
```
Comment (kou) [inline c_glib/plasma-glib/plasma-client.cpp]: Can you change to `gplasma_client_new(const gchar *store_socket_name, GError **error)`?

```cpp
GPlasmaClient *
gplasma_client_new(const gchar *store_socket_name, GError **error)
{
  auto plasma_client = std::make_shared<plasma::PlasmaClient>(); 
  auto status = client->Connect(store_socket_name, "");
  // ...
}
```

We will add more constructors for `manager_socket_name`, `release_delay` and `num_retries` in `PlasmaClient::Connect()` later.

File: c_glib/test/plasma/test-plasma-client.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# specific language governing permissions and limitations
+# under the License.
+
+class TestPlasmaClient < Test::Unit::TestCase
+  def setup
+    omit("Plasma is required") unless defined?(::Plasma)
+  end
+
+  def test_new
+    assert_nothing_raised do
```
Comment (kou) [inline c_glib/test/plasma/test-plasma-client.rb]: We can test connecting to store server by running `plasma_store_server`.
We can get `plasma_store_server` path from `plasma.pc`: `pkg-config --variable=executable plasma`

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
This is a good start point!

File: c_glib/plasma-glib/plasma-client.h (hunk 1 above)
Comment (shiro615) [reply c_glib/plasma-glib/plasma-client.h]: I'll rename it.

File: c_glib/plasma-glib/plasma-client.cpp (hunk 2 above)
Comment (shiro615) [reply c_glib/plasma-glib/plasma-client.cpp]: OK! I'll change to `gplasma_client_new(const gchar *store_socket_name, GError **error)`.

**Comment by shiro615** [comment]
Thank you for review. I'll try to address them.

File: c_glib/test/plasma/test-plasma-client.rb (hunk 3 above)
Comment (shiro615) [reply c_glib/test/plasma/test-plasma-client.rb]: I’ve added test cases. https://github.com/apache/arrow/pull/2848/commits/5c7a85024f33b171d52757eb19f9353d6f85523d#diff-8c8187dfbb3bc22f4933d0622e440b7d
Please confirm my understanding.

**Review by kou — APPROVED** [review]
+1
I'll merge when CI is green.

**Comment by kou** [comment]
Merged. Thanks!

File: c_glib/test/plasma/test-plasma-client.rb (hunk 3 above)
Comment (shiro615) [reply c_glib/test/plasma/test-plasma-client.rb]: Thanks!

**Pull request closed without merging**